lib/_http_server.js

```diff
@@ -217,6 +217,11 @@ function writeHead(statusCode, reason, obj) {
     // This class of status code indicates a provisional response,
     // consisting only of the Status-Line and optional headers, and is
     // terminated by an empty line.
+
+    if(statusCode == 304 && (headers.hasOwnProperty('trailer') || headers.hasOwnProperty('Trailer')){
```

Comment:
At this point the header properties *should* be lowercase. There shouldn't be a reason to check for `'Trailer'`.